onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -171,6 +171,90 @@ activation.)DOC")
       })
       .SetDoc(R"DOC(Tokenizer divides each string in X into a vector of strings along the last axis. All input strings including attributes are UTF-8 encoded.)DOC");
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(Ngram)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "X", "Input for n-gram extraction", "T")
+      .Output(0, "Y", "NGram results", "T1")
+      .TypeConstraint(
+          "T",
+          {"tensor(string)", "tensor(int32)", "tensor(int64)"},
+          "Input is ether string UTF-8 or int32/int64")
+      .TypeConstraint(
+          "T1",
+          {"tensor(float)"},
+          "1-D tensor of floats")
+      .Attr(
+          "N",
+          "Maximum n-gram length",
+          AttributeProto::INT)
+      .Attr(
+          "M",
+          "Minimum n-gram length",
+          AttributeProto::INT)
+      .Attr(
+          "all",
+          "Boolean to store all n-gram lengths [M-N] or only N.",
+          AttributeProto::INT)
+      .Attr(
+          "S",
```

Comment:
>S [](start = 11, length = 1)

Meaningful name

Reply:
Personally, an attribute's name is always not meaningful enough, and its actual meaning comes from its doc string. For example, if you remove the doc string of scikit-learn's [Tfidf](https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.TfidfVectorizer.html) but still keep attribute names the same, you will see that no user can understand that ops. Also, having short attribute name could make cross-attribute documentation easier and cleaner.